src/cairoWriter/writers/sourceUnitWriter.ts

```diff
@@ -41,21 +45,54 @@ export class SourceUnitWriter extends CairoASTNodeWriter {
       writer.write(v),
     );
 
-    const functions = node.vFunctions.map((v) => writer.write(v));
+    const [importFunctions, generatedFunctions, functions] = node.vFunctions.reduce(
+      ([importFunctions, generatedFunctions, functions], funcDef) =>
+        funcDef instanceof CairoImportFunctionDefinition
+          ? [[funcDef, ...importFunctions], generatedFunctions, functions]
+          : funcDef instanceof CairoGeneratedFunctionDefinition
+          ? [importFunctions, [funcDef, ...generatedFunctions], functions]
+          : [importFunctions, generatedFunctions, [funcDef, ...functions]],
+      [
+        new Array<CairoImportFunctionDefinition>(),
+        new Array<CairoGeneratedFunctionDefinition>(),
+        new Array<FunctionDefinition>(),
+      ],
+    );
```

Comment:
That's a `filter`

Reply:
Something like this is what you have in mind?
```ts
   const importFunctions = node.vFunctions.filter(
      (f) => f instanceof CairoImportFunctionDefinition,
    );
    const generatedFunctions = node.vFunctions.filter(
      (f) => f instanceof CairoGeneratedFunctionDefinition,
    );
    const functions = node.vFunctions.filter(
      (f) =>
        !(f instanceof CairoGeneratedFunctionDefinition) &&
        !(f instanceof CairoImportFunctionDefinition),
    );
```

Reply:
Yes